tslearn/matrix_profile.py

```diff
@@ -71,15 +71,21 @@ class MatrixProfile(TransformerMixin,
     Parameters
     ----------
     subsequence_length : int (default: 1)
-        Length of the subseries (also called window size) to be used for 
+        Length of the subseries (also called window size) to be used for
         subseries distance computations.
-    
+
+    algorithm : str (default: None)
+        Algorithm to use to compute the matrix profile.
+        Defaults to None to use the tslearn implementation.
```

Comment:
Maybe use "numpy" for the base, pure numpy, implementation?

Reply:
Would it make sense to add a note as to when a user might want to use STUMPY instead of the default?  The primary take home message is speed/performance and memory optimization in STUMPY. 

Essentially, we want to avoid the situation(s) where the user thinks that this transformer is too slow or too memory intensive and decides not to use it. Instead, it may be helpful to detect this (say, by checking the size of the time series or the memory consumption) and warn the user or provide clear options. Just a thought.

Reply:
Yes, I think the doc should be clear that `stumpy`-based implementations are expected to be faster and less memory consuming and we could have a link to stumpy docs.